### ISSUE TITLE
How to index/search fields through where in lowercase

### ISSUE THREAD

**Issue by zenzei** [ankane/searchkick#1270]
Hello! Searchkick seems a fantastic gem, thanks for creating it!

I have one question though, we always need to find through "where" condition lowercased. But as i read on this comment https://github.com/ankane/searchkick/issues/177#issuecomment-39811519
_Search is case-insensitive. where option is case-sensitive (just like SQL)._

Is there a way to have the "where" behave as case-insensitive globally? There was a mention on the same issue to remove the "token" filter, wouldn't that go in the opposite direction?
https://github.com/ankane/searchkick/issues/177#issuecomment-340422886

Basically i want to avoid having to downcase every field on the _search_data_ method, and on every term when doing the search, as that will be prone to errors.

Another thing that would like to know if there is a way to have "where" perform like a sql LIKE query. There was an issue created long time ago, not sure if some support was added since then
https://github.com/ankane/searchkick/issues/305

Thanks again for this wonderful gem

**Comment by ankane** [comment]
Hey @zenzei, you can achieve 1) with custom mappings (see Advanced Search docs), but it'll be more complex/harder to maintain than downcasing everything. For 2) you can pass a regex as a where condition (see Querying docs).

**Comment by zenzei** [comment]
Thanks for your answer @ankane.

I must say i'm not a pro in elasticsearch, but i tried using something like 
` Product.search body: {query: {match: {name: "milk"}}}`

Unfortunately that doesn't seem to be case sensitive, i don't get results if the text indexed was Milk or MILK (checking the index in fact text was not stored lowercased).

Of course i'm trying to avoid storing the same fields duplicated on both variantes (normal and lowercased).

Regarding the regex, i tried that and works, though that maybe there was some configuration so i can force that to be automatic. The thing is i think regex doesn't work with "match" on the query. 

Maybe i'm just using it wrong, what's the recommended way to use searchkick for doing? Manually downcasing all field on search_data for indexing, and then manually downcasing all text before doing the search, and not useing"match" as regex are not supported there?

**Comment by ankane** [comment]
`where` conditions are case-sensitive. Unfortunately, I don't have the bandwidth to help with app-specific issues, so please use Stack Overflow if you need additional help.

**Comment by zenzei** [comment]
No problem i understand, thought the case insensitive was a common problem but didn't found proper answers on previous issues here.
Thanks again for your help!
